### DISCUSSION TITLE
Workshop 1

### DISCUSSION THREAD

**Discussion by wenbinyu-PU** [wenbinyugroup/GitBranchingDemo#14] [General]
what does "waiting for CI and approval of other developers" mean? CI stands for what? Does this mean that I created a pull request, I need to wait for others to approve my request.

**Comment by banghuazhao** [comment]
CI stands for Continuous Integration. It is a set of commands to run for pull requests. These commands include but not limited to run compile the source code, run unit tests, and show warnings/errors. In this repo, we can see the detail of CI:

<img width="965" alt="Screenshot 2024-03-02 at 12 14 02 PM" src="https://github.com/wenbinyugroup/GitBranchingDemo/assets/19228705/ed88aeae-30fa-4c35-8644-9c7a8df9f263">

Here is a quick start for how to setup GitHub CI: https://docs.github.com/en/actions/quickstart

For waiting for approval of other developers, this pull request needs approval of others before it can be merged. We can set up rules like if we need approval of other developers or how many approval we need before a pull request can be merged.

<img width="1268" alt="Screenshot 2024-03-02 at 12 05 56 PM" src="https://github.com/wenbinyugroup/GitBranchingDemo/assets/19228705/1d75789f-7075-4ea2-90c2-b58b2e7808b3">
